cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/publisher/MetadataKafkaTestBase.java

```diff
@@ -82,12 +82,16 @@
 
   @BeforeClass
   public static void setup() throws IOException {
+    int kafkaPort = Networks.getRandomPort();
+    Preconditions.checkState(kafkaPort > 0, "Failed to get random port.");
     int zkServerPort = Networks.getRandomPort();
     zkServer = InMemoryZKServer.builder().setDataDir(TMP_FOLDER.newFolder()).setPort(zkServerPort).build();
     zkServer.startAndWait();
-    kafkaServer = new EmbeddedKafkaServer(generateKafkaConfig());
+    kafkaServer = new EmbeddedKafkaServer(generateKafkaConfig(kafkaPort));
     kafkaServer.startAndWait();
     cConf = CConfiguration.create();
+    cConf.set(Constants.Metadata.UPDATES_PUBLISH_ENABLED, "true");
+    cConf.set(Constants.Metadata.KAFKA_BROKER_LIST, "127.0.0.1:" + kafkaPort);
```

Comment:
can use ``InetAddress.getLoopbackAddress().getHostAddress()`` instead of hard coding